testutils/resourcebuilder/machine/v1beta1/aws_provider_spec.go

```diff
@@ -127,26 +169,124 @@ func (m AWSProviderSpecBuilder) BuildRawExtension() *runtime.RawExtension {
 	}
 }
 
+// WithAMI sets the AMI for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithAMI(ami machinev1beta1.AWSResourceReference) AWSProviderSpecBuilder {
+	m.ami = &ami
+	return m
+}
+
 // WithAvailabilityZone sets the availabilityZone for the AWS machine config builder.
 func (m AWSProviderSpecBuilder) WithAvailabilityZone(az string) AWSProviderSpecBuilder {
-	m.availabilityZone = az
+	m.availabilityZone = &az
+	return m
+}
+
+// WithBlockDevices sets the BlockDevices for the AWS machine config builder.
+// Use WithBlockDevices(nil), to set them to the zero value.
+func (m AWSProviderSpecBuilder) WithBlockDevices(blockDevices []machinev1beta1.BlockDeviceMappingSpec) AWSProviderSpecBuilder {
+	m.blockDevices = &blockDevices
+	return m
+}
+
+// WithCredentialsSecret sets the DeviceIndex for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithDeviceIndex(deviceIndex int64) AWSProviderSpecBuilder {
+	m.deviceIndex = &deviceIndex
+	return m
+}
+
+// WithCredentialsSecret sets the CredentialsSecret for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithCredentialsSecret(credentialsSecret *corev1.LocalObjectReference) AWSProviderSpecBuilder {
+	m.credentialsSecret = &credentialsSecret
+	return m
+}
+
+// WithIAMInstanceProfile sets the AMI for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithIAMInstanceProfile(iamInstanceProfile *machinev1beta1.AWSResourceReference) AWSProviderSpecBuilder {
+	m.iamInstanceProfile = &iamInstanceProfile
 	return m
 }
 
 // WithInstanceType sets the instanceType for the AWS machine config builder.
 func (m AWSProviderSpecBuilder) WithInstanceType(instanceType string) AWSProviderSpecBuilder {
-	m.instanceType = instanceType
+	m.instanceType = &instanceType
+	return m
+}
+
+// WithKeyName sets the keyName for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithKeyName(keyName *string) AWSProviderSpecBuilder {
+	m.keyName = &keyName
+	return m
+}
+
+// WithLoadBalancers sets the LoadBalancers for the AWS machine config builder.
+// Use WithLoadBalancers(nil), to set them to the zero value.
+func (m AWSProviderSpecBuilder) WithLoadBalancers(loadBalancers []machinev1beta1.LoadBalancerReference) AWSProviderSpecBuilder {
+	m.loadBalancers = &loadBalancers
+	return m
+}
+
+// WithMetadataServiceOptions sets the MetadataServiceOptions for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithMetadataServiceOptions(opts machinev1beta1.MetadataServiceOptions) AWSProviderSpecBuilder {
+	m.metadataServiceOptions = &opts
+	return m
+}
+
+// WithSpotMarketOptions sets the SpotMarketOptions for the AWS machine config builder.
+func (m AWSProviderSpecBuilder) WithSpotMarketOptions(opts *machinev1beta1.SpotMarketOptions) AWSProviderSpecBuilder {
```

Comment:
Should come later, after SecurityGroups?

Reply:
TY